src/api/Api.js

```diff
@@ -229,14 +229,27 @@ class Api {
             requestUrl = requestUrl.substr(0, requestUrl.indexOf('?'));
         }
 
+        // Encode existing query parameters, since tomcat does not accept unencoded brackets. Throw
+        // an error if they're already encoded to prevent double encoding.
+        if (query) {
+            const isEncoded = query !== decodeURIComponent(query);
+
+            if (isEncoded) {
+                return Promise.reject(
+                    new Error('Cannot process URL encoded URLs, pass an unencoded URL'),
```

Comment:
There's an extraneous "URL" in the message (the first one)

Reply:
I meant to express: `Cannot process URL-encoded URLs, pass an unencoded URL`, so cannot process it if it's been url-encoded. I thought to write it without hyphen as that's how I saw it being used most, but I see it doesn't read that well. Is it clear with the hyphen you think, or shall I reword it?

Reply:
Yeah, I think the hyphen is needed in this case.

Reply:
Cool, added it!